ij-plugin/src/main/kotlin/org/jetbrains/research/tasktracker/config/scenario/models/Scenario.kt

```diff
@@ -58,6 +40,25 @@ data class Scenario(
         return step
     }
 
+    @Suppress("ReturnCount")
+    fun getNextUnit(project: Project): ScenarioUnit? {
+        if (currentStepIterator?.hasNext() != true) {
```

Comment:
I'd suggest to rewrite it a bit:
```kotlin
    private fun Iterator<ScenarioUnit>?.notNullAndHasNext() = this?.hasNext() ?: false

    @Suppress("ReturnCount")
    fun getNextUnit(project: Project): ScenarioUnit? {
        if (!currentStepIterator.notNullAndHasNext()) {
            return null
        }

        cleanStepSettings()
        val currentStep = getNextStep() ?: return null
        currentStep.prepareSettings(project)
        currentStepIterator = currentStep.getUnits().iterator()
        return if (currentStepIterator.notNullAndHasNext()) {
            currentStepIterator?.next()
        } else {
            null
        }
    }
```

what do you think?

Reply:
good idea!